assets/js/components/Dashboard/NewPaMessage/StationsAndZones/SelectZonesPage.tsx

```diff
@@ -0,0 +1,541 @@
+            <div className={`route ${ROUTE_TO_CLASS_NAMES_MAP["Green"]}`}>
+              {placeBranches.join(" ")}
+            </div>
+          )}
+          <div className="place-name">{place.name}</div>
+          {route === "Bus" && (
+            <div className={`route ${ROUTE_TO_CLASS_NAMES_MAP["Bus"]}`}>
+              Bus
+            </div>
+          )}
+        </div>
+      </td>
+      <td className="cell all-button-cell">
+        <Button
+          className={
+            allSignsSelected ? "button-primary" : "button-primary-outline"
+          }
+          onClick={() => {
+            const signIDsAtPlace = allSignsForRouteAtPlace.map((s) => s.id);
+            if (allSignsSelected) {
+              setSignIds(fp.without(signIDsAtPlace, allSelectedSigns));
+            } else {
+              setSignIds(
+                fp.concat(
+                  allSignsForRouteAtPlace.map((s) => s.id),
+                  allSelectedSigns,
+                ),
+              );
+            }
+          }}
+        >
+          All
+        </Button>
+      </td>
+      <td className="cell left-cell">
+        <div className="sign-button-group">
+          {signsInZones(LEFT_ZONES).map((sign) => {
+            return (
+              <SelectSignButton
+                key={sign.id}
+                onClick={() => {
+                  onSignButtonClick(sign.id);
+                }}
+                isSelected={allSelectedSigns.includes(sign.id)}
+              >
+                <ArrowLeftShort /> {sign.label ?? getZoneLabel(sign.zone ?? "")}
+              </SelectSignButton>
+            );
+          })}
+        </div>
+      </td>
+      <td className="cell middle-cell">
+        <div className="sign-button-group">
+          {signsInZones(MIDDLE_ZONES).map((sign) => {
+            return (
+              <SelectSignButton
+                key={sign.id}
+                onClick={() => {
+                  onSignButtonClick(sign.id);
+                }}
+                isSelected={allSelectedSigns.includes(sign.id)}
+              >
+                {sign.label ?? getZoneLabel(sign.zone ?? "")}
+              </SelectSignButton>
+            );
+          })}
+        </div>
+      </td>
+      <td className="cell right-cell">
+        <div className="sign-button-group">
+          {signsInZones(RIGHT_ZONES).map((sign) => {
+            return (
+              <SelectSignButton
+                key={sign.id}
+                onClick={() => {
+                  onSignButtonClick(sign.id);
+                }}
+                isSelected={allSelectedSigns.includes(sign.id)}
+              >
+                {sign.label ?? getZoneLabel(sign.zone ?? "")}{" "}
+                <ArrowRightShort />
+              </SelectSignButton>
+            );
+          })}
+        </div>
+      </td>
+    </tr>
+  );
+};
+
+interface SelectSignButtonProps {
+  isSelected: boolean;
+  onClick: () => void;
+  children: ReactNode;
+}
```

Comment:
**suggestion:** if you need a `children` prop i'd suggest using `React.PropsWithChildren`

```suggestion
type SelectSignButtonProps = React.PropsWithChildren<{
  isSelected: boolean;
  onClick: () => void;
}>;
```